jvm-runtime/ftl-runtime/common/runtime/src/main/java/xyz/block/ftl/runtime/CallRequestContext.java

```diff
@@ -0,0 +1,56 @@
+package xyz.block.ftl.runtime;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+
+import io.quarkus.arc.Arc;
+import io.smallrye.context.SmallRyeThreadContext;
+import io.smallrye.context.api.ThreadContextConfig;
+import xyz.block.ftl.v1.Metadata;
+
+@RequestScoped
+public class CallRequestContext {
```

Comment:
@RequestScoped is already an implicit call request context. I would just create a `CurrentTransaction` request scoped class to track the transaction id.